File: Week 6/sentimental-readability/readability.py
# Input from the user
text = str(input("Text: "))

# Variables that we'll use later
letters = sentences = 0
words = 1

# For every letter inside text, see if we sum the words, letters or
# sentences
for i in range(0, len(text)):
    aux = text[i]
    if aux == " ":
        words += 1
    elif aux == '.' or text[i] == '!' or text[i] == '?':
        sentences += 1
    elif aux.isalpha():
        letters += 1

# Calculates the L and S from Coleman-Liau index
L = ((letters * 1.0) / (words)) * 100
S = ((sentences * 1.0) / (words)) * 100

cl_index = round((0.0588 * L) - (0.296 * S) - 15.8)

# Prints the right value
if (cl_index < 1):
    print(f"Before Grade 1")
elif (1 <= cl_index <= 15):
    print(f"Grade {cl_index}")
else:
    print(f"Grade 16+")

File: Week 2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Get text input from user and initialize counters
    string text = get_string("Text: ");
    int len = strlen(text);
    int counter = 0, sentences = 0, words = 1;

    // Loop through each character in the text
    for (int i = 0; i < len; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
        else if (isalpha(text[i]))
        {
            counter++;
        }
    }

    // Calculate the Coleman-Liau index by determining L and S as
    // L = average number of letters per 100 words
    // S = average number of sentences per 100 words
    float L = ((counter * 1.0) / words) * 100;
    float S = ((sentences * 1.0) / words) * 100;
    int index = round(((0.0588 * L) - (0.296 * S) - 15.8));

    // Print the grade level based on the index
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 1 && index <= 15)
    {
        printf("Grade %i\n", index);
    }
    else
    {
        printf("Grade 16+\n");
    }
    return 0;
}
